src/main.cpp

```diff
@@ -173,23 +173,23 @@ R"(Empirical programming language
                     << std::endl;
           line.clear();
         }
-      }
-
-      // eval and print
-      try {
-        VVM::Timer timer;
-        std::string res = eval(line, true, dump_ast, dump_hir, dump_vvm);
-        std::cout << res << std::endl;
-        if (!res.empty()) {
-          std::cout << std::endl;
+      } else {
```

Comment:
This `else` prevents the timer from running.

What I expect:

```
>>> \t 1 + 1
2

 0ms
```

Reply:
I see, let me have another look.

Reply:
Did you consider a consistent timer setting with something like `set timing on/off`?

Reply:
I wanted to make it similar to both iPython and q/kdb+. They both allow individualized timings in the REPL, which means that I can perform work before and afterwards in a clean REPL and only get timings for the one line I want to check.

Reply:
Slight change: All `\` commands except `\t` clear the line => Check that the line is non empty before eval'ing.

Reply:
The reason I omitted the empty-line check is so that I could print everything consistently:

```
Empirical version 0.1.0-9-g6927177
Copyright (C) 2019 Empirical Software Solutions, LLC

>>> 

>>> 1
1

>>> \l some_code.emp

>>> factorial(5)
120

>>> 

>>> 
```

By not evaluating an empty line, only some of the user's entries have a space printed underneath:

```
Empirical version 0.1.0-10-g7412189
Copyright (C) 2019 Empirical Software Solutions, LLC

>>> 
>>> 1
1

>>> \l some_code.emp
>>> factorial(5)
120

>>> 
>>> 
```

I made a deliberate choice to always have a blank space appear before each new prompt. (Readability is a personal preference, so there's no objective answer here.)

By the way, I agree with your choice to move the `eval()` to the `try` block.

Reply:
This feels like bike-shedding by now but I made yet another update to get the newline but keep the `if` to prevent this from happening:

```
➜  empirical-lang git:(REPLlogic) ✗ ./build/empirical --dump-vvm --dump-ast --dump-hir           
Empirical version 0.1.0-10-g7412189                                                                                                                                                          
Copyright (C) 2019 Empirical Software Solutions, LLC                                                                                                                                         
                                                                                                       
>>>                                                                                                      
0x55e0da738930 Module([], )                                                                                               
                                                                                                            
0x55e0da729510 Module([], )                                                                                  
                                                                                                     
                                                                                                          
                                                                                          
                                                                                                                                                                                             
                                                                                                                                                                                             
                                                                                                                                                                                             
>>>  
```